Build failure: python313Packages.realtime

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Just build and test are failing @Siegema 

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/305927825

### Relevant log output

```console
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
unpacking source archive /nix/store/8kzs5i1ka447n987n1aldwh8qljvligc-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/usage.py"
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built realtime-2.5.2-py3-none-any.whl
Finished creating a wheel...
/build/source/dist /build/source
Unpacking to: unpacked/realtime-2.5.2...OK
Repacking wheel as ./realtime-2.5.2-py3-none-any.whl...OK
/build/source
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for realtime-2.5.2-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
Executing pypaInstallPhase
Successfully installed realtime-2.5.2-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
shrinking RPATHs of ELF executables and libraries in /nix/store/95amgkwaaq2bk77gbxkqq47fsnm3x1p8-python3.13-realtime-py-2.5.2
checking for references to /build/ in /nix/store/95amgkwaaq2bk77gbxkqq47fsnm3x1p8-python3.13-realtime-py-2.5.2...
patching script interpreter paths in /nix/store/95amgkwaaq2bk77gbxkqq47fsnm3x1p8-python3.13-realtime-py-2.5.2
stripping (with command strip and flags -S -p) in  /nix/store/95amgkwaaq2bk77gbxkqq47fsnm3x1p8-python3.13-realtime-py-2.5.2/lib
shrinking RPATHs of ELF executables and libraries in /nix/store/s2wn6phi103z91j41gh9yxpj6zkxsk3g-python3.13-realtime-py-2.5.2-dist
checking for references to /build/ in /nix/store/s2wn6phi103z91j41gh9yxpj6zkxsk3g-python3.13-realtime-py-2.5.2-dist...
patching script interpreter paths in /nix/store/s2wn6phi103z91j41gh9yxpj6zkxsk3g-python3.13-realtime-py-2.5.2-dist
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: pythonCatchConflictsPhase
Running phase: pythonRemoveBinBytecodePhase
Running phase: pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: realtime
Running phase: pytestCheckPhase
Executing pytestCheckPhase
pytest flags: -m pytest
============================= test session starts ==============================
platform linux -- Python 3.13.6, pytest-8.4.1, pluggy-1.6.0
rootdir: /build/source
configfile: pyproject.toml
collected 17 items                                                             

tests/test_connection.py .FFFF                                           [ 29%]
tests/test_presence.py F.....                                            [ 64%]
tests/test_timer.py FFFFFF                                               [100%]

=================================== FAILURES ===================================
____________________________ test_broadcast_events _____________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
____________________________ test_postgrest_changes ____________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
________________________ test_multiple_connect_attempts ________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
________________________ test_send_message_reconnection ________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
________________________________ test_presence _________________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
__________________________ test_timer_initialization ___________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
_____________________________ test_timer_schedule ______________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
_______________________________ test_timer_reset _______________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
________________________ test_timer_multiple_schedules _________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
__________________________ test_timer_callback_error ___________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
___________________________ test_timer_cancellation ____________________________
async def functions are not natively supported.
You need to install a suitable plugin for your async framework, for example:
  - anyio
  - pytest-asyncio
  - pytest-tornasync
  - pytest-trio
  - pytest-twisted
=============================== warnings summary ===============================
../../nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/config/__init__.py:1474
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/config/__init__.py:1474: PytestConfigWarning: Unknown config option: asyncio_default_fixture_loop_scope
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

../../nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/config/__init__.py:1474
  /nix/store/l9wxarvas1jpan9s1l0k1a5mx05g3qj6-python3.13-pytest-8.4.1/lib/python3.13/site-packages/_pytest/config/__init__.py:1474: PytestConfigWarning: Unknown config option: asyncio_mode
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

tests/test_connection.py:64
  /build/source/tests/test_connection.py:64: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_connection.py:104
  /build/source/tests/test_connection.py:104: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_connection.py:247
  /build/source/tests/test_connection.py:247: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_connection.py:306
  /build/source/tests/test_connection.py:306: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_presence.py:26
  /build/source/tests/test_presence.py:26: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_timer.py:12
  /build/source/tests/test_timer.py:12: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_timer.py:24
  /build/source/tests/test_timer.py:24: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_timer.py:44
  /build/source/tests/test_timer.py:44: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_timer.py:65
  /build/source/tests/test_timer.py:65: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_timer.py:88
  /build/source/tests/test_timer.py:88: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

tests/test_timer.py:104
  /build/source/tests/test_timer.py:104: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.asyncio

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_connection.py::test_broadcast_events - Failed: async def functions are not natively supported.
FAILED tests/test_connection.py::test_postgrest_changes - Failed: async def functions are not natively supported.
FAILED tests/test_connection.py::test_multiple_connect_attempts - Failed: async def functions are not natively supported.
FAILED tests/test_connection.py::test_send_message_reconnection - Failed: async def functions are not natively supported.
FAILED tests/test_presence.py::test_presence - Failed: async def functions are not natively supported.
FAILED tests/test_timer.py::test_timer_initialization - Failed: async def functions are not natively supported.
FAILED tests/test_timer.py::test_timer_schedule - Failed: async def functions are not natively supported.
FAILED tests/test_timer.py::test_timer_reset - Failed: async def functions are not natively supported.
FAILED tests/test_timer.py::test_timer_multiple_schedules - Failed: async def functions are not natively supported.
FAILED tests/test_timer.py::test_timer_callback_error - Failed: async def functions are not natively supported.
FAILED tests/test_timer.py::test_timer_cancellation - Failed: async def functions are not natively supported.
================== 11 failed, 6 passed, 13 warnings in 0.23s ===================
```

### Additional context

This causes failure for multiples package which depends on this

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.42, NixOS, 25.05 (Warbler), 25.05.808723.b1b329146965`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/nixpentest/.config/nix/nix.conf:/nix/store/xw3j08zld8g13j8cjackamrpgyv7a4vf-plasma-workspace-6.3.6/etc/xdg/nix/nix.conf:/nix/store/4agsnjbgzj5kl5vs7bnxh58zshsi1h0j-kglobalacceld-6.3.6/etc/xdg/nix/nix.conf:/nix/store/mvj65igdzb8dkyfc82dmnnvh86xakh3p-baloo-6.14.0/etc/xdg/nix/nix.conf:/home/nixpentest/.config/kdedefaults/nix/nix.conf:/nix/store/xw3j08zld8g13j8cjackamrpgyv7a4vf-plasma-workspace-6.3.6/etc/xdg/nix/nix.conf:/nix/store/4agsnjbgzj5kl5vs7bnxh58zshsi1h0j-kglobalacceld-6.3.6/etc/xdg/nix/nix.conf:/nix/store/mvj65igdzb8dkyfc82dmnnvh86xakh3p-baloo-6.14.0/etc/xdg/nix/nix.conf:/home/nixpentest/.config/kdedefaults/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/nixpentest/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/nixpentest/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/nixpentest/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/cfymv98909cdrdgjvp8lrkkws397fqq8-lix-2.93.3/share`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/3g2x3mf4r9as5qavpmca33vighh9gfj9-source`

### Notify maintainers


@Siegema 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/postgrest/default.nix
+++ b/pkgs/development/python-modules/postgrest/default.nix
@@ -12,7 +12,7 @@
 }:
 
 buildPythonPackage rec {
-  pname = "postgrest-py";
+  pname = "postgrest";
   version = "1.1.1";
   pyproject = true;
 

--- a/pkgs/development/python-modules/realtime/default.nix
+++ b/pkgs/development/python-modules/realtime/default.nix
@@ -3,41 +3,40 @@
   buildPythonPackage,
   fetchFromGitHub,
   poetry-core,
-  python-dateutil,
+  pydantic,
   typing-extensions,
   websockets,
   aiohttp,
+  pytest-asyncio,
   pytestCheckHook,
   python-dotenv,
 }:
 
 buildPythonPackage rec {
   pname = "realtime-py";
-  version = "2.5.2";
+  version = "2.7.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "supabase";
     repo = "realtime-py";
-    rev = "v${version}";
-    hash = "sha256-NFxWcnt/zpgDehacqK7QlXhmjrh6JoA6xh+sFjD/tt0=";
+    tag = "v${version}";
+    hash = "sha256-cWWgVs+ZNRvBje3kuDQS5L5utkY3z7MluGFNmjf9LFc=";
   };
 
   dependencies = [
-    python-dateutil
+    pydantic
     typing-extensions
     websockets
-    aiohttp
   ];
 
   pythonRelaxDeps = [
     "websockets"
-    "aiohttp"
-    "typing-extensions"
   ];
 
-  # Can't run all the tests due to infinite loop in pytest-asyncio
-  nativeBuildInputs = [
+  nativeCheckInputs = [
+    aiohttp
+    pytest-asyncio
     pytestCheckHook
     python-dotenv
   ];
@@ -46,9 +45,11 @@ buildPythonPackage rec {
 
   build-system = [ poetry-core ];
 
+  # requires running Supabase
   doCheck = false;
 
   meta = {
+    changelog = "https://github.com/supabase/realtime-py/blob/${src.tag}/CHANGELOG.md";
     homepage = "https://github.com/supabase/realtime-py";
     license = lib.licenses.mit;
     description = "Python Realtime Client for Supabase";

--- a/pkgs/development/python-modules/supabase/default.nix
+++ b/pkgs/development/python-modules/supabase/default.nix
@@ -16,7 +16,7 @@
 }:
 
 buildPythonPackage rec {
-  pname = "supabase-py";
+  pname = "supabase";
   version = "2.16.0";
   pyproject = true;
 

--- a/pkgs/development/python-modules/supabase/default.nix
+++ b/pkgs/development/python-modules/supabase/default.nix
@@ -4,7 +4,7 @@
   fetchFromGitHub,
   poetry-core,
   gotrue,
-  postgrest-py,
+  postgrest,
   realtime,
   storage3,
   supafunc,
@@ -32,7 +32,7 @@ buildPythonPackage rec {
   pythonRelaxDeps = true;
 
   dependencies = [
-    postgrest-py
+    postgrest
     realtime
     gotrue
     httpx

--- a/pkgs/development/python-modules/supabase/default.nix
+++ b/pkgs/development/python-modules/supabase/default.nix
@@ -12,31 +12,32 @@
   pytestCheckHook,
   python-dotenv,
   pytest-asyncio,
-  pydantic,
 }:
 
 buildPythonPackage rec {
   pname = "supabase";
-  version = "2.16.0";
+  version = "2.17.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "supabase";
     repo = "supabase-py";
     rev = "v${version}";
-    hash = "sha256-n+LVC4R9m/BKID9wLEMw/y/2I589TUXTygSIPfTZwB8=";
+    hash = "sha256-psfDs5BCtUjyPsfLwksNvzLmUKmYDvmxKIDPQE/NmQU=";
   };
 
   build-system = [ poetry-core ];
 
-  propagatedBuildInputs = [
+  # FIXME remove for supabase >= 2.18.0
+  pythonRelaxDeps = true;
+
+  dependencies = [
     postgrest-py
     realtime
     gotrue
     httpx
     storage3
     supafunc
-    pydantic
   ];
 
   nativeBuildInputs = [
@@ -46,8 +47,6 @@ buildPythonPackage rec {
   ];
 
   pythonImportsCheck = [ "supabase" ];
-
-  doCheck = true;
 
   meta = {
     homepage = "https://github.com/supabase/supabase-py";

--- a/pkgs/top-level/python-aliases.nix
+++ b/pkgs/top-level/python-aliases.nix
@@ -516,6 +516,7 @@ mapAliases ({
   Polygon3 = polygon3; # Added 2023-08-08
   posix_ipc = posix-ipc; # added 2024-01-07
   poster3 = throw "poster3 is unmaintained and source is no longer available"; # added 2023-05-29
+  postgrest-py = postgrest; # added 2024-08-29
   postorius = throw "Please use pkgs.mailmanPackages.postorius"; # added 2022-04-29
   powerlineMemSegment = powerline-mem-segment; # added 2021-10-08
   prayer-times-calculator = prayer-times-calculator-offline; # added 2024-08-11

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -11918,7 +11918,7 @@ self: super: with self; {
 
   posix-ipc = callPackage ../development/python-modules/posix-ipc { };
 
-  postgrest-py = callPackage ../development/python-modules/postgrest-py { };
+  postgrest = callPackage ../development/python-modules/postgrest { };
 
   posthog = callPackage ../development/python-modules/posthog { };
 

